new quickstart guides (#3060)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/createproject.md

> **Note:** The sample is compatible with iOS 14 and later.

Launch Xcode and create a new iOS app project using SwiftUI for the interface and Swift for the language.

#### Add the UI

> The main view implements all of the app functionality, including initializing the Okta SDK. Note that best practice for a full app uses a singleton to manage the interaction with the SDK.

Reviewer: ```suggestion
The main view implements all of the app functionality, including initializing the Okta SDK. Note that the best practice for a full app uses a singleton to manage the interaction with the SDK.
```

Reviewer 2: Good catch :-)

> Text views display the current state and any requested information, such as the access token. The available buttons depend on the sign in state of the user. Update the imported modules and replace the `ContentView` `struct` with the following code:

Reviewer: ```suggestion
Text views display the current state and any requested information, such as the access token. The available buttons depend on the sign-in state of the user. Update the imported modules and replace the `ContentView` `struct` with the following code:
```

    }
    .alert(isPresented: $showingError) {
      Alert(title: Text(errorTitle ?? "Unknown Error"),
      message: Text(errorMessage ?? "An unknow error occured."))
    }
    .onAppear() {
      configureSDK()
    }
  }

  func signIn() {
    statusText = "Sign in button touched."
    signedIn = true
  }

  func signOut() {
    statusText = "Sign out button touched."
    signedIn = false
  }

  func showUserInfo() {
    statusText = "Show user info button touched"
  }

  func showTokenInfo() {
    statusText = "Show token info button touched"
  }

  func refreshToken() {
    statusText = "Refresh token button touched"
  }

  func configureSDK() {
    print("configureSDK called")
  }

  func tokenExpired() {
    print("tokenExpired called")
  }

  func showError(title: String , message: String) {
    errorTitle = title
    errorMessage = message
    showingError = true
  }

  func showError(title: String, error: Error) {
    showError(title: title, 
    message: error.localizedDescription)
  }

  func updateStatus(_ statusText: String, infoText: String, signedInStatus: Bool) {
    self.statusText = statusText
    self.infoText = infoText
    signedIn = signedInStatus
  }

}
```

> The last four functions are utilities. You'll fill out `tokenExpired` later. The two variants of `showError` configure the content,s and then show an alert (the `showingError` state variable controls the presentation of the alert.)

Reviewer: ```suggestion
The last four functions are utilities. You fill out `tokenExpired` later. The two variants of `showError` configure the contents and then show an alert (the `showingError` state variable controls the presentation of the alert).
```

> The last one updates the status text, the info text, and the `signedIn` state variable which controls the buttons presented at the top of the UI.

Reviewer: ```suggestion
The last one updates the status text, the info text, and the `signedIn` state variable that controls the buttons presented at the top of the Admin Console.
```

#### Add an extension to find the root view

Redirect sign-in is designed for apps using view controllers and storyboards. Some of the calls include a parameter for the current view controller, something that doesn't exist in SwiftUI. This sample uses the root view controller for the first key window of the active app for the parameter.

> Add a new Swift file to your project called `UIApplication+rootViewController` and add the following code:

Reviewer 2: I think that sentence needs a lead in:

To find the root view controller, add a new...

Author: updated
